app/jobs/update_diaper_partner_job.rb

```diff
@@ -1,12 +1,13 @@
 # Creates a job for indicating that a Partner has been invited, and notifies the PartnerBase system about them
-class UpdateDiaperPartnerJob
-  include Sidekiq::Worker
-  include DiaperPartnerClient
-
+class UpdateDiaperPartnerJob < ApplicationJob
   def perform(partner_id)
     @partner = Partner.find(partner_id)
     @invitation_message = @partner.organization.invitation_text
     @response = DiaperPartnerClient.post(@partner.attributes, @invitation_message) if Flipper.enabled?(:email_active)
-    @partner.invited!
+    if @response == Net::HTTPSuccess
+      @partner.invited!
+    else
+      @partner.error!
```

Comment:
Previously we just blanket-set the partner to "invited" status, but there was a test indicating it should sometimes be set to "error"